tests/unit/completion/test_histcategory.py

```diff
@@ -241,3 +241,13 @@ def test_timestamp_fmt(fmt, expected, model_validator, config_stub, init_sql):
     model_validator.set_model(cat)
     cat.set_pattern('')
     model_validator.validate([('foo', '', expected)])
+
+
+def test_skip_duplicate_set(message_mock, caplog, hist):
+    cat = histcategory.HistoryCategory()
+    cat.set_pattern('foo')
+    with caplog.at_level(logging.DEBUG):
```

Comment:
Don't think this should be needed - debug messages are captured in tests by default.